### PR TITLE
Remove explicit GitHub dependency from Gemfile

### PR DESCRIPTION
**Pull request by marcotc**
I noticed that every time I ran `bundle install` in this repository, Bundler would take a second or two to fetch this gem:
```
$ bundle install
Fetching https://github.com/nixme/pry-nav.git
```

It seems like [we introduced this GitHub dependency](https://github.com/DataDog/dd-trace-rb/pull/310/files#diff-8b7db4d5cc4b8f6dc8feb7030baa2478R5-R6) due to an incompatibility between our `pry`, debugger `pry-nav`, and the version of `pry` we used at the time.

Fast-forward a year, a new version `pry-nav` has been released that supports `pry` version `< 13`.

I've also unlocked the patch version from `pry-stack_explorer`, as there was a patch release since we last locked it.

The new versions are compatible with all our supported versions of Ruby.

### CODE REVIEW

File: Appraisals [hunk 1]
```diff
@@ -305,15 +305,15 @@ elsif Gem::Version.new('2.2.0') <= Gem::Version.new(RUBY_VERSION) \
     appraise 'rails5-postgres-redis' do
       gem 'rails', '~> 5.2.1'
       gem 'pg', '< 1.0', platform: :ruby
-      gem 'redis'
+      gem 'redis', '~> 4'
```
Comment (marcotc) [inline Appraisals]: Shaking up our depencies has revealed that we are not requiring the minimum version of `redis` required by [Rails 5.2](https://github.com/rails/rails/blob/v5.2.0/activesupport/lib/active_support/cache/redis_cache_store.rb#L8).
This allowed `bundler` to install `redis ~> 3` in CI, which is not compatible with the version of Rails under test.

Comment (marcotc) [reply Appraisals]: Moved to its own PR: https://github.com/DataDog/dd-trace-rb/pull/890

**Comment by marcotc** [comment]
Rails CI issue has been fixed. PR is ready for review.

**Comment by marcotc** [comment]
That's a good point @ericmustin. Now that JRuby support is fully merged I tried the `pry` development experience again, and I noticed that `pry-stack_explorer` installs but doesn't work correctly. I couldn't find any alternatives, so I just removed it from installing with JRuby for the time being.

**Comment by marcotc** [comment]
Build will fail because of Rails 6.1.

**Review by ericmustin — APPROVED** [review]
👍

**Pull request merged**